src/Ocelot.Provider.Kubernetes/OcelotBuilderExtensions.cs

```diff
@@ -9,10 +11,23 @@ public static class OcelotBuilderExtensions
     public static IOcelotBuilder AddKubernetes(this IOcelotBuilder builder, bool usePodServiceAccount = true)
     {
         builder.Services
-            .AddKubeClient(usePodServiceAccount)
+            .AddSingleton<IKubeApiClient, KubeApiClient>(KubeApiClientFactory)
```

Comment:
**It appears to be a breaking change 🤯 Lovely singletons once more❗**
The main purpose of the `Ocelot.Provider.Kubernetes` NuGet package is to facilitate the integration of the `KubeClient` package with `Ocelot`. The proposed changes undermine this objective by removing `AddKubeClient`. This design direction is flawed because our solutions should be developed based on the official `KubeClient` package's functionality, not by altering the library's behavior to suit our needs!
You have to understand that the issue is not `AddKubeClient` extension, but singleton factory aka `KubernetesProviderFactory.Get` with bad scope management!

Reply:
> It appears to be a breaking change 🤯

Why is it breaking change?
✔️  Public API hasn't changed
✔️  IKubeApiClient has the same singleton lifetime as it had before this fix(implicitly promoted)
Proposed change - make IKubeApiClient singleton lifetime explicit by registering it as singleton.

> our solutions should be developed based on the official KubeClient package's functionality, not by altering the library's behavior to suit our needs!

`KubeClient` package's functionality - talking to kube api as httpclient wrapper. I don't understand why it's lifetime can't be managed by `Ocelot.Provider.Kubernetes`. (like it was done for [consul](https://github.com/ThreeMammals/Ocelot/blob/ce0227c059f220761acc2d880e554174bdb3c544/src/Ocelot.Provider.Consul/ConsulClientFactory.cs)).